pulsaradmin/pkg/utils/topic_auto_creation_config.go

```diff
@@ -20,5 +20,5 @@ package utils
 type TopicAutoCreationConfig struct {
 	Allow      bool      `json:"allowAutoTopicCreation"`
 	Type       TopicType `json:"topicType"`
-	Partitions int       `json:"defaultNumPartitions"`
+	Partitions *int      `json:"defaultNumPartitions"`
```

Comment:
it's not allowed to pass the `partitions` argument when the type is `non-partitioned`, so need to set this as a pointer